src/Controllers/SolutionsController.cs

```diff
@@ -76,23 +78,57 @@ public async Task<IActionResult> PutSolution(Guid id, Solution solution)
         // POST: api/Solutions/article
         // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
         [HttpPost("article")]
-        public async Task<ActionResult<Solution>> PostArticle(Article solution)
+        public async Task<ActionResult<Solution>> PostArticle(ArticleRequestModel article)
         {
-            _context.Solutions.Add(solution);
+            //convert to datamodel
+            Article dataModel = ArticleRequestModelConverter.ConvertReqModelToDataModel(article);
+
+            //check existence of SDGs
+            foreach(var item in dataModel.SDGs)
+            {
+                if (!SDGExists(item.SDGId))
+                {
+                    return NotFound();
+                }
+            }
+
+            //ensure many to many relationship is properly defined
+            await PopulateManyToManySDGs(dataModel);
+
+            //save entity
+            _context.Solutions.Add(dataModel);
             await _context.SaveChangesAsync();
 
-            return CreatedAtAction("GetSolution", new { id = solution.Id }, solution);
+            //return http response
+            return CreatedAtAction("GetSolution", new { id = dataModel.Id }, dataModel);
         }
 
         // POST: api/Solutions/howto
         // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
         [HttpPost("howto")]
-        public async Task<ActionResult<Solution>> PostHowTo(HowTo solution)
+        public async Task<ActionResult<Solution>> PostHowTo(HowToRequestModel howTo)
         {
-            _context.Solutions.Add(solution);
+            //convert to datamodel
+            HowTo dataModel = HowToRequestModelConverter.ConvertReqModelToDataModel(howTo);
+
+            //check existence of SDGs
+            foreach (var item in dataModel.SDGs)
+            {
+                if (!SDGExists(item.SDGId))
+                {
+                    return NotFound();
+                }
+            }
+
+            //ensure many to many relationships are properly defined
+            await PopulateManyToManySDGs(dataModel);
```

Comment:
Use _context.AttachRange(howTo.SDGs) to do this

Reply:
https://docs.microsoft.com/en-us/ef/core/saving/related-data

Dat hoeft niet